clients/StateChannelWallet.ts

```diff
@@ -0,0 +1,54 @@
+import { ethers } from 'ethers';
+import { UserOperation } from './types';
+
+export class StateChannelWallet {
+  private intermediaryUrl: string;
+  private chainProvider: ethers.providers.JsonRpcProvider;
+  private signer: ethers.Wallet;
+
+  constructor(params: { signingKey: string; chainRpcUrl: string; intermediaryUrl: string }) {
+    this.intermediaryUrl = params.intermediaryUrl;
+
+    this.chainProvider = new ethers.providers.JsonRpcProvider(params.chainRpcUrl);
+
+    const wallet = new ethers.Wallet(params.signingKey);
+    this.signer = wallet.connect(this.chainProvider);
+  }
+
+  async getCurrentBlockNumber(): Promise<number> {
+    try {
+      const blockNumber = this.chainProvider.getBlockNumber();
+      return blockNumber;
+    } catch (error: any) {
+      throw new Error(`Error fetching block number: ${error.message}`);
+    }
+  }
+
+  async signUserOperation(userOp: UserOperation): Promise<UserOperation> {
+    // Serialize the UserOperation into bytes
+    const serialized = ethers.utils.concat([
+      ethers.utils.arrayify(userOp.sender),
+      ethers.utils.arrayify(ethers.BigNumber.from(userOp.nonce)),
+      userOp.initCode,
+      userOp.callData,
+      ethers.utils.arrayify(ethers.BigNumber.from(userOp.callGasLimit)),
+      ethers.utils.arrayify(ethers.BigNumber.from(userOp.verificationGasLimit)),
+      ethers.utils.arrayify(ethers.BigNumber.from(userOp.preVerificationGas)),
+      ethers.utils.arrayify(ethers.BigNumber.from(userOp.maxFeePerGas)),
+      ethers.utils.arrayify(ethers.BigNumber.from(userOp.maxPriorityFeePerGas)),
+      userOp.paymasterAndData
+    ]);
+
+    // keccak256 hash the serialized bytes then sign the hash
+    const hash = ethers.utils.keccak256(serialized);
+    const signature = await this.signer.signMessage(ethers.utils.arrayify(hash));
+
+    // Append the signature to the user operation
+    userOp.signature = ethers.utils.arrayify(signature);
```

Comment:
Since we expect the `UserOperation.signature` to contain [two 65 byte signatures](https://github.com/statechannels/nitro-smart-contract-wallet/blob/7a0ae38cd6272f7ca2f091118fdcb4713d7a932d/contracts/Nitro-SCW.sol#L61) packed together, we probably don't want to set`UserOp.Signature`  when signing it. We'll probably want to call `signUserOperation` for both owner and intermediary and  [set signature to the concatenated signatures](https://github.com/statechannels/nitro-smart-contract-wallet/blob/7a0ae38cd6272f7ca2f091118fdcb4713d7a932d/test/Nitro-SCW.ts#L54)

Reply:
You are correct. But I will forgo fixing this code and instead use the existing code you referenced earlier that performs this same operation